.github/workflows/update-infrastructure-repo.yaml

```
@@ -149,6 +154,96 @@ jobs:
           number: ${{ steps.cpr.outputs.pull-request-number }}
           repo: ${{ inputs.source-github-repo }}
 
+      - name: Append prior PR bodies to current one
+        if: ${{ inputs.preserve-closed-pr-bodies}}
+        env:
+          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
+          REPO: ${{ github.server_url }}/${{ inputs.source-github-repo }}
+          CURRENT_PR_NUMBER: ${{ steps.cpr.outputs.pull-request-number }}
+          SERVICE: ${{ inputs.service }}
+          ENVIRONMENT: ${{ matrix.environment }}
+        run: |
+          set -eu -o pipefail
+          set -x
+
+          # Only run when a real service name is set
+          if [[ -z "${SERVICE}" || "${SERVICE}" == "service" ]]; then
+            echo "No explicit service set, skipping append of prior PR bodies."
+            exit 0
+          fi
+
+          workdir="$(mktemp -d)"
+          trap 'rm -rf "$workdir"' EXIT
+
+          BODY_LIMIT=65000 # hard cap to stay under GitHub's PR body limit
+
+          # Fetch current PR body
+          gh pr view "$CURRENT_PR_NUMBER" --repo "$REPO" --json body --jq .body > "$workdir/current_body.md" || echo "" > "$workdir/current_body.md"
```

Nitpick. Something like this may help avoid the small duplication. I haven't tested it though, and there's probably a better way to default to an empty string.

```suggestion
          current_body=$(gh pr view "$CURRENT_PR_NUMBER" --repo "$REPO" --json body --jq .body || echo "")
          echo "$current_body" > "$workdir/current_body.md"
```